modules/zookeeper/src/main/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoveryImpl.java

```diff
@@ -242,11 +242,11 @@ public ZookeeperDiscoveryImpl(
         DiscoverySpiListener lsnr,
         DiscoverySpiDataExchange exchange,
         IgniteDiscoverySpiInternalListener internalLsnr,
-        ZookeeperDiscoveryStatistics stats) {
+        ZookeeperDiscoveryStatistics stats,
+        JdkMarshaller marsh
+    ) {
         assert locNode.id() != null && locNode.isLocal() : locNode;
 
-        MarshallerUtils.setNodeName(marsh, igniteInstanceName);
```

Comment:
can we remove MarshallerUtils.setNodeName from public access and move it possibly to IgniteKernal ?

Reply:
Removing anything from the public API is a compatibility violation.